mitodl/open-discussions
# Cc/smaller bundle

#### Pre-Flight checklist


#### What are the relevant tickets?
#3714 

#### What's this PR do?
1.  This PR reduces the bundle size of open discussions and infinite corridor:
    ```
                            master      this branch
    ------------------------------------------------                        
    Open Discussions:       7.5M        4.2M
    Infinite Corridor:      5.8M        2.5M
    ```
2. Additionally, added [webpack-bundle-analyzer](https://www.npmjs.com/package/webpack-bundle-analyzer) to both webpack configs so that we can more easily see what is contributing to our bundle size.

**How does this reduce bundle size?**
- Primarily, this reduces the bundle size by removing the factory functions from our local modules's top-level exports. So:
```ts
// Before:
import { makeLearningResource} from "ol-search-ui"
// Now:
import { makeLearningResource} from "ol-search-ui/build/factories"
``` 
See https://github.com/mitodl/open-discussions/issues/3714 for more.

#### How should this be manually tested?
1. In the `watch` container, run `NODE_ENV=production yarn run build`.
2. Run `ls -lh frontends/infinite-corridor/build` and `ls -lh frontends/open-discussions/build`.
    - Check that the JS bundles are smaller like I said. Check that there is
    - Check that there is no report.html file
3. In the container again, run `WEBPACK_ANALYZE=True NODE_ENV=production yarn run build`.
4. Open `frontends/infinite-corridor/build/report.html` in your browser to see the bundle analysis.

#### Screenshots 

This is the infinite corridor bundle report... There's still some work we could do, e.g., CKEditor is only used by moderators, I think it could be loaded dynamically. But that can be looked into separately.

<img width="1683" alt="Screen Shot 2022-09-07 at 10 26 30 AM" src="https://user-images.githubusercontent.com/9010790/188903505-29231852-d3bb-44ad-b27e-a055b6668cb6.png">
